pkg/webhook/servicecatalog/clusterservicebroker/validation/handler_access.go

```diff
@@ -0,0 +1,118 @@
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"context"
+	"fmt"
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/webhookutil"
+	authenticationapi "k8s.io/api/authentication/v1"
+	authorizationapi "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// AccessToBroker handles ClusterServiceBroker validation
+type AccessToBroker struct {
+	decoder *admission.Decoder
+	client  client.Client
+}
+
+// InjectDecoder injects the decoder
+func (h *AccessToBroker) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+// InjectClient injects the client
+func (h *AccessToBroker) InjectClient(c client.Client) error {
+	h.client = c
+	return nil
+}
+
+// Validate checks if client has access to cluster service broker if broker requires authentication
+// This feature was copied from Service Catalog admission plugin https://github.com/kubernetes-incubator/service-catalog/blob/v0.1.41/plugin/pkg/admission/broker/authsarcheck/admission.go
+// If you want to track previous changes please check there.
+func (h *AccessToBroker) Validate(ctx context.Context, req admission.Request, csb *sc.ClusterServiceBroker, traced *webhookutil.TracedLogger) error {
+	if csb.Spec.AuthInfo == nil {
+		traced.Infof("%s %q has no AuthInfo. Operation completed", csb.Kind, csb.Name)
+		return nil
+	}
+
+	var secretRef *sc.ObjectReference
+	if csb.Spec.AuthInfo.Basic != nil {
+		secretRef = csb.Spec.AuthInfo.Basic.SecretRef
+	} else if csb.Spec.AuthInfo.Bearer != nil {
+		secretRef = csb.Spec.AuthInfo.Bearer.SecretRef
+	}
+
+	if secretRef == nil {
+		traced.Infof("%s %q has no SecretRef neither in Basic nor Bearer auth. Operation completed", csb.Kind, csb.Name)
+		return nil
+	}
+
+	user := req.UserInfo
+	sar := &authorizationapi.SubjectAccessReview{
+		Spec: authorizationapi.SubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationapi.ResourceAttributes{
+				Namespace: secretRef.Namespace,
+				Verb:      "get",
+				Group:     corev1.SchemeGroupVersion.Group,
+				Version:   corev1.SchemeGroupVersion.Version,
+				Resource:  corev1.ResourceSecrets.String(),
+				Name:      secretRef.Name,
+			},
+			User:   user.Username,
+			Groups: user.Groups,
+			Extra:  convertToSARExtra(user.Extra),
+			UID:    user.UID,
+		},
+	}
+
+	err := h.client.Create(ctx, sar)
+	if err != nil {
+		traced.Errorf("Could not create SubjectAccessReview for %s %q: %v", csb.Kind, csb.Name, err)
+		return nil
+	}
+
+	if !sar.Status.Allowed {
+		traced.Infof(
+			"Could not handle %s operation for %s %q because SubjectAccessReview has allowed status set to false",
+			req.Operation,
+			csb.Kind,
+			csb.Name,
+		)
+		return fmt.Errorf("could not %s %s %q", req.Operation, csb.Kind, csb.Name)
```

Comment:
please return the same error msg as it was originally. Your error does not say what was the problem.